homework/index.js

```diff
@@ -30,15 +31,111 @@
     return elem;
   }
 
-  function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
-      } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
-      }
+  function reposAbstract(index, data) {
+    const repo = data[index];
+    const leftDiv = document.querySelector('.leftDiv');
+    const table = createAndAppend('table', leftDiv, { class: 'table' });
+    const tBody = createAndAppend('tbody', table);
+    const repositoryRow = createAndAppend('tr', tBody, { class: 'row' });
+    createAndAppend('td', repositoryRow, { text: 'Repository :', class: 'label' });
+    const repositoryRowSecondTd = createAndAppend('td', repositoryRow);
+    createAndAppend('a', repositoryRowSecondTd, {
+      href: repo.html_url,
+      target: '_blank',
+      text: repo.name,
+    });
+    const descriptionRow = createAndAppend('tr', tBody, { class: 'row' });
+    if (repo.description !== null) {
+      createAndAppend('td', descriptionRow, {
+        text: 'Description :',
+        class: 'label',
+      });
+      createAndAppend('td', descriptionRow, { text: repo.description });
+    }
+
```

Comment:
As a style choice you could have put created a function to create a row.  This would help with not having to repeat the same code for each row.  So the code would look something like this
e.g. 
```
const descriptionRow = appendRow(tBody, 'Description', repo.description);
const forksRow = appendRow(tBody, 'Forks', repo.forks);
const dateRow = appendRow(tBody, 'Updated', new Date(repo.updated_at).toLocaleString())
```
You might have to do some magic in `appendRow` to allow the appending of HTML elements, but the first element which is an `<a/>` tag could be done manually as you do in your code above.
This would still make things a lot shorter and more readable, and if something needed adjusting it could be done in one place.

Reply:
you are right it is shorter and readable. I will change it.